### ISSUE TITLE
Support @ts-ignore comments in template markup

### ISSUE THREAD

**Issue by madeleineostoja** [sveltejs/language-tools#1026]
**Is your feature request related to a problem? Please describe.**
Since svelte can't (yet?) support typescript expressions in component template markup, it's impossible to provide typings/coercions/etc in template logic blocks. This results in seas of red warnings and errors in the editor and svelte-check.

**Describe the solution you'd like**
Assuming giving Svelte full typescript support in markup is either not going to happen or not happening any time soon, it would at least be nice to support `@ts-ignore` HTML comments to skip typescript warnings for the following line of code. 

**Describe alternatives you've considered**
The only real alternative is to disable typescript diagnostics in language-tools entirely, which is obviously less than ideal

**Comment by dummdidumm** [comment]
This is tricky if not impossible for some situations, for example if you have your attributes spread out over multiple lines. `@ts-ignore` only works for the next line. What is the sea of errors you talk about specifically? Can this be solved by more explicit typing up front?

**Comment by madeleineostoja** [comment]
Hmm that’s fair, I hadn’t thought of attributes spreading over multiple lines. That would make @ts-ignore comments fragile and unintuitive. 

Happy to just close this and hope for type support in template markup instead

**Comment by lovasoa** [comment]
I use this nasty trick to disable type checking of a single attribue : 

```svelte
<script lang="ts">
	const notypecheck = (x:any)=>x;
</script>

<MyComponent {...notypecheck({
    attribute:  value
}) />
```

**Comment by rogadev** [comment]
> I use this nasty trick to disable type checking of a single attribute :
> 
> ```svelte
> <script lang="ts">
> 	const notypecheck = (x:any)=>x;
> </script>
> 
> <MyComponent {...notypecheck({
>     attribute:  value
> }) />
> ```

How would you do that for something like this:
```svelte
{#each columns as blok}
  <div class="flex-auto px-6">
    <StoryblokComponent {blok} /> <!-- (property) StoryblokComponentProps.blok: SbBlokData
                                                                 Type 'string' is not assignable to type 'SbBlokData'.  -->
  </div>
{/each}
```

And if a method of doing so exists, is it easier than:

```svelte
{#each columns as blok}
  <div class="flex-auto px-6">
    <!-- @ts-ignore -->
    <StoryblokComponent {blok} />
  </div>
{/each}
```

**Comment by dummdidumm** [comment]
If this would be implemented, it would be through a `<!-- @ts-ignore -->` comment. Since comments cannot be placed within the tag itself, the question is what the more intuitive/wanted behavior here is:
```html
<!-- @ts-ignore -->
<ComponentThatHasError
  {propThatHasError}
/>
```
should the error from `propThatHasError` also be ignored, or only the error from `<ComponentThatHasError`? My feeling is that everything inside the component should be ignored, because
- no other way to silence through a comment
- no depending on how things are formatted, this is inconsistent: if `propThatHasError` is on the same line, that error is silenced, too, else it isn't

Thoughts welcome

**Comment by madeleineostoja** [comment]
That would certainly be the intuitive behavior, but it also feels fragile? I don’t know implementation details though.

I guess the wider question is has there been any momentum on allowing ts in template expressions? That would make this issue kind of moot

**Comment by dummdidumm** [comment]
Good point, with TS inside template expressions you could do casting instead, which would probably solve this for 90% of use cases. What's left would be things like silencing errors like "don't know this component" or "this component/element doesn't have prop X", which are probably rare.

**Comment by crushingCodes** [comment]
I couldn't get the suggestion above to work, but luckily the component was my own, so i was able to get around it using the following:

MyComponent.svelte
```
<script lang="ts">
	import type { IThing } from '$lib/server/types';

	export let value: unknown;

	const thing: IThing = value as IThing;
</script>
```

Consumer.svelte
```
<MyComponent value={value} />
```

**Comment by LorenzoBloedow** [comment]
I have a suggestion: we could prefix attributes with `ts-ignore-` or just use the directive `ts-ignore-all` to ignore the entire component.

**Comment by nikonikoniko** [comment]
I have managed to sneak in a `@ts-ignore` in like this:

```svelte
<Input
	type="file"
	multiple
	{...attrs}
	{...{
		...{} 
		/* @ts-ignore */
	}}
	bind:files={$formData.file}
/>
```

I used this to replace the useful trick from @lovasoa

**Comment by chaychoong** [comment]
I tried this, seems to work

```
{/* @ts-ignore */ null}
<label htmlFor="example">Example</label>
```

**Comment by danbord** [comment]
> I tried this, seems to work
> 
> ```
> {/* @ts-ignore */ null}
> <label htmlFor="example">Example</label>
> ```

Thanks for that!

**Comment by M-erb** [comment]
> I tried this, seems to work
> 
> ```
> {/* @ts-ignore */ null}
> <label htmlFor="example">Example</label>
> ```

Was exactly what I was looking for! What is the compiler looking for that it needs that null there though?

**Comment by StevenBee95** [comment]
> > I tried this, seems to work
> > ```
> > {/* @ts-ignore */ null}
> > <label htmlFor="example">Example</label>
> > ```
> 
> Was exactly what I was looking for! What is the compiler looking for that it needs that null there though?

Maybe it just needs to find some kind of expression inside curly bracket and "null" just happens to be not doing anything

**Comment by eric-blue** [comment]
> > > I tried this, seems to work
> > > ```
> > > {/* @ts-ignore */ null}
> > > <label htmlFor="example">Example</label>
> > > ```
> > 
> > 
> > Was exactly what I was looking for! What is the compiler looking for that it needs that null there though?
> 
> Maybe it just needs to find some kind of expression inside curly bracket and "null" just happens to be not doing anything

without the `null` the file breaks the compile

for what it's worth, this also works

```
{/* @ts-expect-error -- your reason here */ null}
<Component {...props}>
```

**Comment by dummdidumm** [comment]
Now that Svelte 5 supports TS syntax inside the template, how big of a deal is this still? I assume a lot of the thumbs up where added before you were able to e.g. do `{(x as any).y}` in the template.

**Comment by dummdidumm** [comment]
As of https://github.com/sveltejs/language-tools/pull/2950 you can ignore type errors of attributes/props
